src/index.ts

```diff
@@ -238,7 +237,12 @@ export interface Purpose {
   canonicalPurposeCode?: string
   legalBasisName?: string
   legalBasisDescription?: string
+
+  // the identifier of the stack that the purpose belongs to when displayed. If empty, this purpose belongs to no stack
```

Comment:
this feels back to front. wouldn't the stack list the purpose codes in order?

Reply:
Yeah that works. Then keep track if a purpose is not a member of a stack and display those below the stack?

Reply:
Only those listed in stacks are shown, right? Or is the definition that any not in a stack are shown at a different level? Or do we have a "default" stack?